pkgs/development/libraries/openmpi/default.nix

```diff
@@ -10,41 +10,46 @@
 with stdenv.lib;
 
 let
-  majorVersion = "1.10";
+  majorVersion = "3.0";
+  minorVersion = "0";
 
 in stdenv.mkDerivation rec {
-  name = "openmpi-${majorVersion}.7";
+  name = "openmpi-${majorVersion}.${minorVersion}";
 
   src = fetchurl {
     url = "http://www.open-mpi.org/software/ompi/v${majorVersion}/downloads/${name}.tar.bz2";
-    sha256 = "142s1vny9gllkq336yafxayjgcirj2jv0ddabj879jgya7hyr2d0";
+    sha256 = "1mw2d94k6mp4scg1wnkj50vdh734fy5m2ygyrj65s4mh3prbz6gn";
   };
 
-  buildInputs = [ gfortran ]
-    ++ optional (stdenv.isLinux || stdenv.isFreeBSD) libibverbs;
+  postPatch = ''
+    patchShebangs ./
+  '';
+
+  buildInputs = with stdenv; [ gfortran zlib ]
+    ++ optional isLinux libnl
+    ++ optional (isLinux || isFreeBSD) libibverbs;
 
   nativeBuildInputs = [ perl ];
 
   configureFlags = []
+    ++ optional stdenv.isLinux  "--with-libnl=${libnl.dev}"
     ++ optional enableSGE "--with-sge"
     ++ optional enablePrefix "--enable-mpirun-prefix-by-default"
     ;
 
   enableParallelBuilding = true;
 
-  preBuild = ''
-    patchShebangs ompi/mpi/fortran/base/gen-mpi-sizeof.pl
-  '';
-
   postInstall = ''
-		rm -f $out/lib/*.la
+    rm -f $out/lib/*.la
    '';
 
+  doCheck = true;
+
   meta = {
     homepage = http://www.open-mpi.org/;
     description = "Open source MPI-2 implementation";
     longDescription = "The Open MPI Project is an open source MPI-2 implementation that is developed and maintained by a consortium of academic, research, and industry partners. Open MPI is therefore able to combine the expertise, technologies, and resources from all across the High Performance Computing community in order to build the best MPI library available. Open MPI offers advantages for system and software vendors, application developers and computer science researchers.";
-    maintainers = [ ];
+    maintainers = with maintainers; [ markuskowa ];
```

Comment:
It should be `with stdenv.lib.maintainers` I think.

This would be fine if in line 48 you had `meta = with stdenv.lib; {` (which is often done since `stdenv.lib` is used to access `licenses` and `maintainers`). Since no license is given here (maybe there should), you can just fix the `with` in line 52.